client-app/shared/static-content/components/call-to-action.vue

```diff
@@ -14,15 +19,32 @@
 </template>
 
 <script setup lang="ts">
-defineProps({
-  model: {
-    type: Object,
-    required: true,
-  },
+import type { RouteLocationRaw } from "vue-router";
 
-  settings: {
-    type: Object,
-    required: true,
-  },
-});
+type ButtonType = {
+  label: string;
+  link: string;
+};
+
+interface IProps {
+  id: string;
+  background?: string;
+  title?: string;
+  buttons?: Array<ButtonType>;
+}
+
+defineProps<IProps>();
+
+type LinkAttrType = { to: RouteLocationRaw } | { link: string } | object;
+
+const getLinkAttr = (link?: string): LinkAttrType => {
+  if (link) {
+    if (link.startsWith("/")) {
+      return { to: link };
+    } else {
+      return { externalLink: link };
+    }
+  }
+  return {};
+};
```

Comment:
We already have exactly the same code in `call-to-action-with-image`, move shared code to composable please